jobapi/socket.go

```diff
@@ -0,0 +1,40 @@
+package jobapi
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/mitchellh/go-homedir"
+)
+
+var random = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+// NewSocketPath generates a path to a socket file (without actually creating the file itself) that can be used with the
+// job api.
+// These files are located in a subdirectory of the home dir (`~/.buildkite-agent/sockets/job-api` on unix machines),
+// and are named after the PID of the process that runs this function, with a random number appended to the end.
```

Comment:
Given then issues we've had with the length of this path in testing, does it make sense to make the PATH to the socket completely configurable? A user with a very long username could find themselves having such issues and may need to find somewhere else to put it. We should give them a heads-up about the 108 character thing too.